### PR TITLE
[pack] Adding Multiple Languages Support in Functions

### PR DESCRIPTION
**Pull request by ghost**
Port of #8449

### CODE REVIEW

File: src/WebJobs.Script/Environment/EnvironmentExtensions.cs [hunk 1]
```diff
@@ -311,14 +314,26 @@ public static string GetAntaresComputerName(this IEnvironment environment)
         }
 
         /// <summary>
-        /// Gets the computer name.
+        /// Gets if runtime enviromnent is logic apps.
```
Comment (madelinegordon) [inline src/WebJobs.Script/Environment/EnvironmentExtensions.cs]: nit - "environment"

File: src/WebJobs.Script/Host/ScriptHost.cs [hunk 2]
```diff
@@ -297,7 +297,8 @@ public async Task InitializeAsync(CancellationToken cancellationToken = default)
                         // Windows Consumption as well.
                         string runtimeVersion = _environment.GetEnvironmentVariable(RpcWorkerConstants.FunctionWorkerRuntimeVersionSettingName);
 
-                        if (!string.IsNullOrEmpty(runtimeVersion))
+                        // If the environemt is multi language, we are running multiple language worker, and thus not honoring 'FUNCTIONS_WORKER_RUNTIME_VERSION'
```
Comment (madelinegordon) [inline src/WebJobs.Script/Host/ScriptHost.cs]: nit - "environment" and "workers"

File: src/WebJobs.Script/Host/ScriptHost.cs [hunk 3]
```diff
@@ -560,7 +568,7 @@ private void AddFunctionDescriptors(IEnumerable<FunctionMetadata> functionMetada
                 var workerConfig = _languageWorkerOptions.Value.WorkerConfigs?.FirstOrDefault(c => c.Description.Language.Equals(_workerRuntime, StringComparison.OrdinalIgnoreCase));
 
                 // If there's no worker config, use the default (for legacy behavior; mostly for tests).
-                TimeSpan initializationTimeout = workerConfig?.CountOptions?.InitializationTimeout ?? WorkerProcessCountOptions.DefaultInitializationTimeout;
+                var initializationTimeout = workerConfig?.CountOptions?.InitializationTimeout ?? WorkerProcessCountOptions.DefaultInitializationTimeout;
```
Comment (madelinegordon) [inline src/WebJobs.Script/Host/ScriptHost.cs]: Just curious - why was this type changed?

Comment (ghost) [reply src/WebJobs.Script/Host/ScriptHost.cs]: Kept it consistent with line above, where `workerConfig` var is used instead of `ClassName`.

Comment (liliankasem) [reply src/WebJobs.Script/Host/ScriptHost.cs]: I think using Timespan makes more sense here. With `workerConfig`, we know the type of the variable, but with `initializationTimeout` it's not clear through assignment what the type is and so it makes sense to be clear with that.

Comment (ghost) [reply src/WebJobs.Script/Host/ScriptHost.cs]: Reverted back, will also backport it to v3.x

**Review by madelinegordon — APPROVED** [review]

**Review by soninaren — APPROVED** [review]

**Pull request merged**